external-crates/move/crates/move-analyzer/src/symbols/mod.rs

```diff
@@ -370,41 +326,78 @@ pub fn compute_symbols_typed_program(
-            let deps_symbols_data = if let Some(cached_symbols_data) = cached_deps.symbols_data {
-                // We have cached results of the dependency symbols computation from the previous run.
-                cached_symbols_data
-            } else {
-                // No cached dependency symbols data but we still have cached compilation results.
-                // Fill out dependency symbols from compiled package info to cache them at the end of analysis
-                let computation_data_deps = run_typing_analysis(
-                    computation_data_deps,
-                    mapped_files,
-                    compiler_info,
-                    &cached_deps.program_deps.typing.modules,
-                );
-                Arc::new(computation_data_deps)
-            };
-            // create `file_use_defs` map and merge references to produce complete symbols data
+    let deps_symbols_data_opt = if let Some(cached_deps) = compiled_pkg_info.cached_deps.clone() {
+        let deps_symbols_data = if let Some(cached_symbols_data) = cached_deps.symbols_data {
+            // We have cached results of the dependency symbols computation from the previous run.
+            // Create `file_use_defs` map and merge references to produce complete symbols data
             // (mod_outer_defs and def_info have already been merged to facilitate user program
-            // analysis)
-            update_file_use_defs(&deps_symbols_data, mapped_files, &mut file_use_defs);
-            for (def_loc, uses) in &deps_symbols_data.references {
+            // analysis).
+            update_file_use_defs(&cached_symbols_data, mapped_files, &mut file_use_defs);
+            for (def_loc, uses) in &cached_symbols_data.references {
                 computation_data
                     .references
                     .entry(*def_loc)
                     .or_default()
                     .extend(uses);
             }
-            Some(deps_symbols_data)
+            cached_symbols_data
         } else {
-            None
+            // No cached dependency symbols which means that dependency symbools
+            // and user-level symbols are all in `computation_data`
+            let dep_mod_ident_strs = computation_data
+                .mod_outer_defs
+                .iter()
+                .filter_map(|(mod_ident_str, mod_defs)| {
+                    if cached_deps.dep_hashes.contains(&mod_defs.fhash) {
+                        Some(mod_ident_str)
+                    } else {
+                        None
+                    }
+                })
+                .collect::<BTreeSet<_>>();
+
+            let deps_computation_data = SymbolsComputationData {
+                mod_outer_defs: computation_data
+                    .mod_outer_defs
+                    .clone()
+                    .into_iter()
+                    .filter(|(mod_ident_str, _)| dep_mod_ident_strs.contains(mod_ident_str))
+                    .collect(),
```

Comment:
Please write a macro that does this iteration

Reply:
Done!